### PR TITLE
Updates CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by janiv**
Documents the process for making changes to the KSL Header including
having all pull requests reviewed and making mention of the build files
and how to generate them.

Also includes notes about how to do a release, update the ghpages, and
information about bumping the version number in package.json.

### CODE REVIEW

**Comment by janiv** [comment]
@ponchorage Can you review this?

**Comment by ponchorage** [comment]
Looks good. Thanks for the clarification and sorry for short-cutting the process. 

What about the static html? When I ran the build script, I didn't see that it generated a new one - just the ksl-header.js and ksl-header-static.js.

Any reason to remove the .idea from the .gitignore file? I use PHPStorm and it creates that directory and we don't want it in the repo.

**Comment by janiv** [comment]
No worries. There was nothing documenting the procedure so it was kind of hard to follow. :)

.idea is specific to your environment and would best belong in your global .gitignore file rather than a project specific one.

https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

If everybody put their environment specific stuff in the each project's .gitignore, you'd things in their for VIM, Sublime, PHPStorm, eclipse, .DS_Store, etc. Those are going to be problems in every single project you work in, different for each user or the repo, and so are best handled in the users' global gitignore file.

**Comment by janiv** [comment]
The static html only changes when it needs to. Doesn't look like any of your changes affected it. If they had, it would have been regenerated in the `gulp build` task.
